drake/solvers/Optimization.h

```diff
@@ -160,19 +245,38 @@ namespace Drake {
       if (Aeq.cols() != A.cols()) throw std::runtime_error("Can't change the number of decision variables");
       A.conservativeResize(Aeq.rows(),Eigen::NoChange);
```

Comment:
Why do this conservatively if we're just going to overwrite it on the next line?